src/Definition.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of DivineNii opensource projects.
+ *
+ * PHP version 7.4 and above required
+ *
+ * @author    Divine Niiquaye Ibok <divineibok@gmail.com>
+ * @copyright 2021 DivineNii (https://divinenii.com/)
+ * @license   https://opensource.org/licenses/BSD-3-Clause License
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Rade\DI;
+
+use Nette\Utils\{Callback, Reflection};
+use PhpParser\Node\{Expr\ArrayDimFetch,
+    Expr\ArrowFunction,
+    Expr\Assign,
+    Expr\New_,
+    Expr\StaticPropertyFetch,
+    Name,
+    Scalar\String_,
+    Stmt\Return_,
+    UnionType};
+use PhpParser\ParserFactory;
+use Rade\DI\Builder\Statement;
+
+/**
+ * Represents definition of standard service.
+ *
+ * @author Divine Niiquaye Ibok <divineibok@gmail.com>
+ */
+class Definition
+{
+    use Traits\ResolveTrait;
+
+    /** Marks a definition as being a factory service. */
+    public const FACTORY = 1;
+
+    /** This is useful when you want to autowire a callable or class string lazily. */
+    public const LAZY = 2;
+
+    /** Marks a definition as being deprecated. */
+    public const DEPRECATED = 3;
+
+    /** Marks a definition as a private service. */
+    public const PRIVATE = 4;
+
+    public ?string $id = null;
+
+    private bool $factory = false;
+
+    private bool $lazy = false;
+
+    private bool $public = true;
+
+    private array $deprecated = [];
+
+    /**
+     * DefinitionValueResolver constructor
+     *
+     * @param mixed                   $entity
+     * @param array<int|string,mixed> $arguments
+     */
+    public function __construct($entity, array $arguments = [], int $type = null)
```

Comment:
Parameter `$type` should be removed, and rather **should** method used instead.

```suggestion
    public function __construct($entity, array $arguments = [])
```